### PR TITLE
Remove utility functions from SolidBaseUpdates

### PR DESCRIPTION
**Pull request by francoishamon**
This PR implements the first cleanup steps towards the merge of #2377:
- Removal of `hyperUpdate`, `hypoUpdate`, `checkSmallStrainStiffness`, and `computeSmallStrainFiniteDifferenceStiffness` from `SolidBaseUpdates`
- Creation of `SolidUtilities` to host the functions above (now free functions)
- Make `saveStress` a `public` function of `SolidBaseUpdates`

I have not modified the implementation of the functions. 
The `smallStrainUpdate` functions are still `virtual`, this will be changed in another PR.

PR ready for review

### CODE REVIEW

File: src/coreComponents/constitutive/solid/ElasticIsotropic.hpp [hunk 1]
```diff
@@ -143,23 +143,6 @@ class ElasticIsotropicUpdates : public SolidBaseUpdates
     return m_shearModulus[k];
   }
 
-  // TODO: confirm hyper stress/strain measures before activatiing
```
Comment (joshua-white) [inline src/coreComponents/constitutive/solid/ElasticIsotropic.hpp:146]: Can we leave the `hyperUpdate` code here, but still commented out?  We will likely activate this in the future, and we don't want to lose the existing implementation from GEOS-classic.  It should not impact this PR to leave it here.

File: src/coreComponents/constitutive/solid/ElasticIsotropic.hpp [hunk 2]
```diff
@@ -309,59 +292,6 @@ void ElasticIsotropicUpdates::smallStrainUpdate( localIndex const k,
   stiffness.m_shearModulus = m_shearModulus[k];
 }
 
-
-// TODO: need to confirm stress / strain measures before activating hyper inferface
```
Comment (joshua-white) [inline src/coreComponents/constitutive/solid/ElasticIsotropic.hpp:313]: Please don't delete.

**Review by joshua-white — APPROVED** [review]
Looks great!

**Review by CusiniM — APPROVED** [review]
Thanks!

File: src/coreComponents/constitutive/solid/ElasticIsotropic.hpp (hunk 1 above)
Comment (francoishamon) [reply src/coreComponents/constitutive/solid/ElasticIsotropic.hpp]: Done, I put the `hyperUpdate` back

File: src/coreComponents/constitutive/solid/ElasticIsotropic.hpp (hunk 2 above)
Comment (francoishamon) [reply src/coreComponents/constitutive/solid/ElasticIsotropic.hpp]: Done

**Pull request merged**